### ISSUE TITLE
[P1] Variant A real-data smoke training + loss-weight ablation

### ISSUE THREAD

**Issue by AcarBasaran** [ahmetnecirvandogan/sketch-generator#52]
**Priority:** P1 — required before paper-claimable Variant A results
**Split off from:** old #43 (the loader-split scope stays in #43; real-data work moved here)
**Depends on:** Stage 1 run on VALAR producing a usable `dataset/metadata.jsonl`; #43 loader split merged; PR #47 (epoch-based trainer) merged

## What this issue covers

Two things that need real data, not just code, and so can only be done after the dataset is generated and the loader split lands.

### Sub-task A — Variant A full real-data smoke training

Goal: confirm Variant A's three-head loss (`L_albedo + λ₁·L_roughness + λ₂·L_lighting`) actually decreases on real data without NaN.

- [ ] Run 100-step Variant A training on `--split train` with real generated samples
- [ ] Log per-component losses (`L_albedo`, `L_roughness`, `L_lighting`) separately
- [ ] Verify all three decrease monotonically (or at least non-divergent) over the 100 steps
- [ ] Verify no NaN in any tensor (loss, gradients, weights)
- [ ] Run one validation pass on `--split val` after step 100 → log per-component val losses

### Sub-task B — Loss-weight ablation

Goal: find sensible defaults for `λ₁` (roughness weight) and `λ₂` (lighting weight). Current values (`λ₁=1.0, λ₂=0.1`) are arbitrary.

- [ ] Small grid over `λ₁ ∈ {0.1, 0.5, 1.0, 2.0}` × `λ₂ ∈ {0.01, 0.1, 0.5, 1.0}`
- [ ] 200-step run for each combination on a subset (e.g. 200 samples)
- [ ] Report final val losses and qualitative inspection of generated maps
- [ ] Pick a default that balances albedo accuracy against the other heads

## Files affected

- `scripts/train_pbr_model.py` (from PR #47) — add per-component loss logging, val pass
- new `scripts/ablation_loss_weights.sh` — driver for the grid search

## How to test

After running:

```bash
python scripts/train_pbr_model.py --variant a --split train --epochs 1 \
    --max-steps 100 --val-every 100 --log-component-losses
```

Output should include `L_albedo`, `L_roughness`, `L_lighting` columns per step.

## When to start

After:
1. `dataset/metadata.jsonl` exists on VALAR (Stage 1 has run with PRs #48 + #51 merged)
2. PR for #43 (loader split) has merged
3. PR #47 (epoch-based trainer) has merged

## Why P1 and not P0

The pipeline can start training with placeholder loss weights `(1.0, 0.1)` and we can observe whether it converges. Sub-task A is essentially the *first end-to-end training run* itself. Sub-task B (ablation) is a refinement and can land after first results are in hand.
